MaxText/max_utils.py

```diff
@@ -360,6 +365,65 @@ def fill_unspecified_mesh_axes(parallelism_vals, target_product, parallelism_typ
   return parallelism_vals
 
 
+def create_custom_64x4_device_mesh(
```

Comment:
Ideally jax should support some type of logic for this, for background this function is copied + modified from the jax internal create_hybrid_device_mesh function.